frontend/src/app/main/data/workspace.cljs

```diff
@@ -1502,6 +1503,123 @@
               :else
               (rx/empty))))))))
 
+(defn copy-selected-css
+  []
+  (ptk/reify ::copy-selected-css
+    ptk/EffectEvent
+    (effect [_ state _]
+      (let [objects (wsh/lookup-page-objects state)
+            selected (->> (wsh/lookup-selected state) (mapv (d/getf objects)))
+            css (css/generate-style objects selected selected {:with-prelude? false})]
+        (wapi/write-to-clipboard css)))))
+
+(defn copy-selected-css-nested
+  []
+  (ptk/reify ::copy-selected-css-nested
+    ptk/EffectEvent
+    (effect [_ state _]
+      (let [objects (wsh/lookup-page-objects state)
+            selected (->> (wsh/lookup-selected state)
+                          (cfh/selected-with-children objects)
+                          (mapv (d/getf objects)))
+            css (css/generate-style objects selected selected {:with-prelude? false})]
+        (wapi/write-to-clipboard css)))))
+
+(defn copy-selected-props
+  []
+  (ptk/reify ::copy-selected-props
+    ptk/WatchEvent
+    (watch [_ state _]
+      (letfn [(fetch-image [entry]
+                (let [url (cf/resolve-file-media entry)]
+                  (->> (http/send! {:method :get
+                                    :uri url
+                                    :response-type :blob})
+                       (rx/map :body)
+                       (rx/mapcat wapi/read-file-as-data-url)
+                       (rx/map #(assoc entry :data %)))))
+
+              (resolve-images
+                [data]
```

Comment:
on letfn, the params vector should be always on the same line as the funcion names symbol, unless multiple arities are defined.

Reply:
ok